src/estimator.c

```diff
@@ -27,75 +27,79 @@ static const vector_t g = {0.0f, 0.0f, -1.0f};
 static vector_t b;
 static quaternion_t q_tilde;
 static quaternion_t q_hat;
-static int32_t last_time;
+static uint64_t last_time;
 
 static bool mat_exp;
 static bool quad_int;
 static bool use_acc;
 
-static float kp_;
-static float ki_;
-static uint32_t init_time;
-
 static vector_t _accel_LPF;
 static vector_t _gyro_LPF;
 
-void init_estimator(bool use_matrix_exponential, bool use_quadratic_integration, bool use_accelerometer)
+void reset_state()
 {
-  _current_state.p = 0.0f;
-  _current_state.q = 0.0f;
-  _current_state.r = 0.0f;
-  _current_state.phi = 0.0f;
-  _current_state.theta = 0.0f;
-  _current_state.psi = 0.0f;
-
-  q_hat.w = 1.0f;
-  q_hat.x = 0.0f;
-  q_hat.y = 0.0f;
-  q_hat.z = 0.0f;
-
-  w1.x = 0.0f;
-  w1.y = 0.0f;
-  w1.z = 0.0f;
-
-  w2.x = 0.0f;
-  w2.y = 0.0f;
-  w2.z = 0.0f;
-
-  b.x = 0.0f;
-  b.y = 0.0f;
-  b.z = 0.0f;
-
-  kp_ = get_param_float(PARAM_FILTER_KP);
-  ki_ = get_param_float(PARAM_FILTER_KI);
-  init_time = get_param_int(PARAM_INIT_TIME)*1000; // microseconds
-
-  w_acc.x = 0.0f;
-  w_acc.y = 0.0f;
-  w_acc.z = 0.0f;
-
-  q_tilde.w = 1.0f;
-  q_tilde.x = 0.0f;
-  q_tilde.y = 0.0f;
-  q_tilde.z = 0.0f;
+    _current_state.q.w = 1.0f;
```

Comment:
I added a reset_state() function so that when we perform an IMU calibration, the state doesn't jump and potentially go unstable.